src/melee.cpp

```diff
@@ -179,8 +210,39 @@ bool player::handle_melee_wear( item &shield, float wear_multiplier )
         g->m.add_item_or_charges( pos(), elem );
     }
 
+    // Preserve item temporarily for component breakdown
+    item temp = shield;
+
     remove_item( shield );
 
+    // Breakdown fragile weapons into components
+    if( temp.has_flag( "FRAGILE_MELEE" ) && !temp.components.empty() ) {
+        add_msg_player_or_npc( m_bad, _( "Your %s breaks apart!" ),
+                                      _( "<npcname>'s %s breaks apart!" ),
+                                      str.c_str() );
+
+        std::vector<item> all_comps = temp.components;
```

Comment:
No need to copy the component vector here. Just iterate over the components themselves.
If you need to make a copy (for example, before wielding), it's better to explicitly copy it there.